### ISSUE TITLE
Core: catch self-interacting pairs?

### ISSUE THREAD

**Issue by GPMueller** [spirit-code/spirit#160]
E.g. if you have a monolayer, but specify a pair with `dc=1`, this will produce self-interacting spins.
Should this be caught and deactivated?

Currently this could be caught in the Fileparser, later this would have to be done by the Hamiltonian.

**Comment by GPMueller** [comment]
When variable geometry is implemented (#102), this will be necessary in order to prevent bugs when reducing the system size..

**Comment by GPMueller** [comment]
#332 was a sort of duplicate of this. The problem has been solved (for now) by changing the function `Vectormath::idx_from_pair` - it now ignores pairs which go beyond the geometry by more than one translation:
```C++
if (std::abs(pair.translations[0]) > Na ||
                std::abs(pair.translations[1]) > Nb ||
                std::abs(pair.translations[2]) > Nc )
                return -1;
```
*Example*: an `a`-`b`-monolayer may therefore still be periodical in `c`-direction, but pairs with `dc >= 2` will be ignored.
